hw_diag/tests/test_download.py

```diff
@@ -0,0 +1,119 @@
+import unittest
+from unittest.mock import patch
+import os
+import responses
+
+from hw_diag.utilities.download import validate_file, download_with_resume
+
+TESTDATA_FILENAME = os.path.join(os.path.dirname(__file__), "data/nebra.webp")
+TESTDATA_SHA256 = "aa3fbe2ddc4ea2ced20ce5d5a34b0f4b618a22bc0656d48a8be2166202d3a00c"
+
+TEST_BLOCK_SIZE = 1024
+
+OUTPUT_FILENAME = "/tmp/nebra.webp"
+
+MOCK_DOWNLOAD_URL = "https://example.com/test.webp"
+MOCK_DOWNLOAD_CONTENT_TYPE = "image/webp"
+
+
+def get_test_data():
+    """
+    Return the test data for download_with_resume
+    """
+    with open(TESTDATA_FILENAME, "rb") as f:
+        data = f.read()
+        return data
+
+
+def add_head_response(data):
+    responses.add(
+        responses.HEAD,
+        MOCK_DOWNLOAD_URL,
+        status=200,
+        content_type=MOCK_DOWNLOAD_CONTENT_TYPE,
+        headers={"Content-Length": f"{len(data)}"},
+    )
+
+
+def add_file_chunk_response(data, start, end):
+    data_len = len(data)
+    responses.add(
+        responses.GET,
+        MOCK_DOWNLOAD_URL,
+        body=data[start:end],
+        status=200,
+        content_type=MOCK_DOWNLOAD_CONTENT_TYPE,
+        headers={
+            "Content-Length": f"{data_len}",
+            "Content-Range": f"bytes {start}-{end}/{data_len}",
+        },
+    )
+
+
+def add_not_found_response():
+    responses.add(
+        responses.GET,
+        MOCK_DOWNLOAD_URL,
+        status=404,
+        content_type=MOCK_DOWNLOAD_CONTENT_TYPE,
+    )
+
+
+class TestDownload(unittest.TestCase):
+    def setUp(self) -> None:
+        self.cleanup()
+        return super().setUp()
+
+    def cleanup(self) -> None:
+        for filename in [OUTPUT_FILENAME, OUTPUT_FILENAME + ".part"]:
+            if os.path.exists(filename):
+                os.remove(filename)
+
+        responses.reset()
+
+    # @patch(requests.get, read_stream)
+    @patch("hw_diag.utilities.download.DEFAULT_BLOCK_SIZE", 1024)
+    @responses.activate
+    def test_download(self):
```

Comment:
@pritamghanghas - Can you explain what this test is testing against? There are a lot of mocking and I'm not sure testing for download would make sense if we're actually mocking the download process?

Reply:
testing that download function will save .part file if it doesn't receive full data. And subsequently finish it later on next call.

Reply:
> @pritamghanghas - Can you explain what this test is testing against? There are a lot of mocking and I'm not sure testing for download would make sense if we're actually mocking the download process?

testing that download function will save .part file if it doesn't receive full data. And subsequently finish it later on next call.